### PR TITLE
Fix environment variable expansion

### PR DESCRIPTION
**Pull request by ulyssessouza**

### CODE REVIEW

**Review by glours — APPROVED** [review]
LGTM, tested against the Compose unit & e2e tests with success!

**Review by nicksieger — APPROVED** [review]
Nice!

File: dotenv/godotenv_var_expansion_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
+package dotenv
```
Comment (milas) [inline dotenv/godotenv_var_expansion_test.go:1]: 👏 for these test cases!

File: dotenv/parser.go [hunk 2]
```diff
@@ -187,6 +196,8 @@ func expandEscapes(str string) string {
 			return "\n"
 		case "r":
 			return "\r"
+		case "$":
+			return "$$"
```
Comment (milas) [inline dotenv/parser.go:200]: 🤑

**Review by milas — APPROVED** [review]
LGTM

**Pull request merged**